internal/cachecontroller/cache_controller.go

```diff
@@ -87,15 +88,23 @@ func (c *InMemoryCacheController) DetermineInvalidation(
 		return entry.LastModified
 	}
 
-	lastModified, err, _ := c.sf.Do(storeID, func() (interface{}, error) {
-		return c.findChangesAndInvalidate(ctx, storeID)
-	})
-
-	if err != nil {
-		return time.Time{}
+	_, present := c.mutexMap[storeID]
+	if !present {
+		c.mutexMap[storeID] = &sync.Mutex{}
 	}
+	hasLock := c.mutexMap[storeID].TryLock()
+	if hasLock {
+		// if the cache cannot be found, we want to invalidate entries in the background
+		// so that it does not block the answer path.
+		go func() {
+			defer c.mutexMap[storeID].Unlock()
```

Comment:
Why not prune the entry? Don't want to carry the entries of low activity stores forever.

Reply:
updated to prune entry